frontend/lib/resolution/can-pass.cpp

```diff
@@ -780,12 +780,12 @@ bool CanPassResult::canInstantiateBuiltin(Context* context,
   if (formalT->isAnyIntegralType() && actualT->isIntegralType())
     return true;
 
-  if (formalT->isAnyIteratorClassType()) {
+  if (actualT->isAnyIteratorClassType()) {
     CHPL_UNIMPL("iterator classes"); // TODO: represent iterators
     return false;
   }
 
-  if (formalT->isAnyIteratorRecordType()) {
+  if (actualT->isAnyIteratorRecordType()) {
```

Comment:
Not sure about this change, can you explain it?

Reply:
Actually, I think I was a bit hasty with this change that was motivated by removing some false positives for the CHPL_UNIMPLEMENTED warnings that these conditions contain. We should still be checking the ``formalT`` here to get correct behavior from ``canPass``.

I'm inclined to remove the CHPL_UNIMPLEMENTED warnings from this section because they don't indicate that iterator records/classes are actually being used, just that we're checking to see if something can be passed to such an argument.

Reply:
That would be OK as well. I would think we could implement them, but I don't think the dyno resolver creates them, and nothing else will before it runs.